runtime/fetcher/fakes_test.go

```diff
@@ -0,0 +1,221 @@
+package fetcher
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"errors"
+	"io"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/taskcluster/taskcluster-worker/runtime/client"
+)
+
+type fakeContext struct {
+	context.Context
+	queue           client.Queue
+	m               sync.Mutex
+	progressReports []float64
+}
+
+func (c *fakeContext) Queue() client.Queue {
+	return c.queue
+}
+
+func (c *fakeContext) Progress(description string, percent float64) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	debug("Progress: %s - %.02f %%", description, percent*100)
+	c.progressReports = append(c.progressReports, percent)
+}
+
+func (c *fakeContext) ProgressReports() []float64 {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.progressReports
+}
+
+type fakeWriteReseter struct {
+	buffer []byte
+}
+
+func (w *fakeWriteReseter) Write(p []byte) (int, error) {
+	w.buffer = append(w.buffer, p...)
```

Comment:
Do we need to expand `p`?

Reply:
yes, because we're appending each byte to an array of bytes... I'm fairly sure golang has optimized this code path :)